server/src/test/java/org/elasticsearch/action/admin/indices/template/get/GetIndexTemplatesResponseTests.java

```diff
@@ -0,0 +1,86 @@
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.template.get;
+
+import org.elasticsearch.cluster.metadata.AliasMetaData;
+import org.elasticsearch.cluster.metadata.IndexTemplateMetaData;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractXContentTestCase;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class GetIndexTemplatesResponseTests extends AbstractXContentTestCase<GetIndexTemplatesResponse> {
+    @Override
+    protected GetIndexTemplatesResponse doParseInstance(XContentParser parser) throws IOException {
+        return GetIndexTemplatesResponse.fromXContent(parser);
+    }
+
+    @Override
+    protected GetIndexTemplatesResponse createTestInstance() {
+        List<IndexTemplateMetaData> templates = new ArrayList<>();
+        int numTemplates = between(0, 10);
+        for (int t = 0; t < numTemplates; t++) {
+            IndexTemplateMetaData.Builder templateBuilder = IndexTemplateMetaData.builder("template-" + t);
+            templateBuilder.patterns(IntStream.range(0, between(1, 5)).mapToObj(i -> "pattern-" + i).collect(Collectors.toList()));
+            int numAlias = between(0, 5);
+            for (int i = 0; i < numAlias; i++) {
+                templateBuilder.putAlias(AliasMetaData.builder(randomAlphaOfLengthBetween(1, 10)));
+            }
+            if (randomBoolean()) {
+                templateBuilder.settings(Settings.builder().put("index.setting-1", randomLong()));
+            }
+            if (randomBoolean()) {
+                templateBuilder.order(randomInt());
+            }
+            if (randomBoolean()) {
+                templateBuilder.version(between(0, 100));
+            }
+            if (randomBoolean()) {
+                try {
+                    templateBuilder.putMapping("doc", "{\"doc\":{\"properties\":{\"type\":\"text\"}}}");
+                } catch (IOException ex) {
+                    throw new UncheckedIOException(ex);
+                }
+            }
+            templates.add(templateBuilder.build());
+        }
+        return new GetIndexTemplatesResponse(templates);
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return false;
```

Comment:
I wonder if it's worth to return true here and exclude some specific paths. for instance at the top-level we can't inject anything arbitrary, within a template it should be possible, within mappings it is not... etc. this is important to verify that we our parsing is forward compatible.

Reply:
For what its worth, should those kind of test be part of the IndexTemplateMetaDataTests instead? The response here just renders a list of IndexTemplateMetaData instances with arbitrary names, so in this level are trying to parse every field name as a IndexTemplateMetaData. The parser in this class will indead fail on anything that it encounters thats not parseable as IndexTemplateMetaData, and at the moment it should I think.

Reply:
IndexTemplateMetaDataTests already covers these situations.

Reply:
ok can you add a comment here then?

Reply:
Yes, I added a comment for this.